lib/ezfile/classes/ezgzipzlibcompressionhandler.php

```diff
@@ -50,13 +50,7 @@ function compressionLevel()
     */
     static function isAvailable()
     {
-        $extensionName = 'zlib';
-        if ( !extension_loaded( $extensionName ) )
-        {
-            $dlExtension = ( eZSys::osType() == 'win32' ) ? '.dll' : '.so';
-            @dl( $extensionName . $dlExtension );
-        }
-        return extension_loaded( $extensionName );
+        return function_exists('gzcompress');
```

Comment:
Okay just to check for one zlib function? or should I check for all of those used in class?

Reply:
(checking existence of function instead of extension due to feedback from @gggeek to support polyfills )

Reply:
@gggeek any opinion on this? I guess if they all are part of same extension, checking one should be ok'ish. The prior code did not take issue of certain functions being disabled manually into account anyway.

Reply:
I'd say: we use up to 3 functions: check for all of them. More than that: check just one...

I cant come up with a 'perfect' plan as each strategy has drawbacks:
- check only for 1 function: we incur the risk that others are not poly-filled
- check for all of them: makes code slow and unreadable
- check for just a couple: makes sense if those are the most-frequently used, eg. they cover 80% of usages by the current class. but also worst of both worlds...